core/src/main/java/io/grpc/ChannelBuilder.java

```diff
@@ -0,0 +1,49 @@
+package io.grpc;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+/**
+* Common base type for all {@link Channel} builders.
+*/
+public abstract class ChannelBuilder<BuilderT extends ChannelBuilder<BuilderT>> {
+  /**
+   * Provides a custom executor.
+   *
+   * <p>It's an optional parameter. If the user has not provided an executor when the channel is
+   * built, the builder will use a static cached thread pool.
+   *
+   * <p>The channel won't take ownership of the given executor. It's caller's responsibility to
+   * shut down the executor when it's desired.
+   */
+  public abstract BuilderT executor(ExecutorService executor);
+
+  /**
+   * Adds interceptors that will be called before the channel performs its real work. This is
+   * functionally equivalent to using
+   * {@link io.grpc.ClientInterceptors#intercept(io.grpc.Channel, java.util.List)},
```

Comment:
This (and below) shouldn't need to be fully-qualified (for ClientInterceptors, Channel, and List). It does need to be in AbstractChannelBuilder because it is now in a different package (although List shouldn't need to be fully qualified in either case).